src/mame/mit/cadr_iob.h

```diff
@@ -0,0 +1,115 @@
+// license:BSD-3-Clause
+// copyright-holders:Wilbert Pol
+#ifndef MAME_MIT_CADR_IOB_H
+#define MAME_MIT_CADR_IOB_H
+
+#pragma once
+
+
+#include "cpu/mcs48/mcs48.h"
+#include "sound/spkrdev.h"
+
+
+DECLARE_DEVICE_TYPE(CADR_IOB, cadr_iob_device)
+
+
+class cadr_iob_device : public device_t
+{
+public:
+	cadr_iob_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	static constexpr feature_type unemulated_features() { return feature::LAN; }
+
+	// configuration
+	auto irq_vector_callback() { return m_irq_vector_cb.bind(); }
+
+	void write(offs_t offset, u16 data);
+	u16 read(offs_t offset);
+	DECLARE_INPUT_CHANGED_MEMBER(mouse_changed);
+
+protected:
+	// device-level overrides
+	virtual void device_start() override ATTR_COLD;
+	virtual void device_reset() override ATTR_COLD;
+	virtual void device_add_mconfig(machine_config &config) override;
+	virtual const tiny_rom_entry *device_rom_region() const override;
+	virtual ioport_constructor device_input_ports() const override;
```

Comment:
Same as the other file – don’t copy-paste comments that don’t make sense, and there are several cold lifecycle member functions here that should be marked cold.